solidity/contracts/BurntPixArchives.sol

```diff
@@ -1,20 +1,28 @@
+// TODO: General Feedback:
+// ReentrancyGuard ? reentrancy issue in IRegistry(registry).refine() and IFractal(fractalClone).refine()?.
+// Nice to have: break down refineToArchive
```

Comment:
@Ledespana break down into separate functions makes the call more costly and it's already expensive. that's why i tried to create the most compact implementation I could think of